### PR TITLE
Data flow: Add `ArgumentNode` consistency checks

### PR DESCRIPTION
**Pull request by hvitved**
Adds two consistency checks for checking uniqueness and existence of `DataFlowCall`s associated to `ArgumentNode`s. While the data flow library can handle `ArgumentNode`s belonging to multiple `DataFlowCall`s, it is good to check that this was indeed the intention. For C#, this would have highlighted https://github.com/github/codeql/pull/14099, and it also revealed four other kinds of inconsistencies ([two](https://github.com/github/codeql/pull/14132) of [which](https://github.com/github/codeql/pull/14170) have been fixed).

### CODE REVIEW

File: ruby/ql/consistency-queries/DataFlowConsistency.ql [hunk 1]
```diff
@@ -35,11 +35,17 @@ private module Input implements InputSig<RubyDataFlow> {
       n.asExpr() = arg
     )
   }
+
+  predicate multipleArgumentCallExclude(ArgumentNode arg, DataFlowCall call) {
+    // An argument such as `x` in `if not x then ...` has two successors (and hence
+    // two calls); one for each Boolean outcome of `x`.
```
Comment (aschackmull) [inline ruby/ql/consistency-queries/DataFlowConsistency.ql:41]: Maybe I just don't know enough about the Ruby modeling, but the `x` in `if not x then ...` doesn't look like an argument to me, and I don't see any calls, so this comment reads a bit cryptic.

File: python/ql/lib/semmle/python/dataflow/new/internal/DataFlowImplConsistency.qll [hunk 2]
```diff
@@ -47,6 +47,10 @@ private module Input implements InputSig<PythonDataFlow> {
   predicate identityLocalStepExclude(Node n) {
     not exists(n.getLocation().getFile().getRelativePath())
   }
+
+  predicate multipleArgumentCallExclude(ArgumentNode arg, DataFlowCall call) {
+    isArgumentNode(arg, call, _)
```
Comment (aschackmull) [inline python/ql/lib/semmle/python/dataflow/new/internal/DataFlowImplConsistency.qll:52]: Maybe add a comment about why this check is excluded?

Comment (hvitved) [reply python/ql/lib/semmle/python/dataflow/new/internal/DataFlowImplConsistency.qll:52]: The Python team (@RasmusWL) are already aware of this, and they will look into it follow-up.

Comment (RasmusWL) [reply python/ql/lib/semmle/python/dataflow/new/internal/DataFlowImplConsistency.qll:52]: (Difference in Python 2 and Python 3 output, even when excluding stdlib :grimacing:)

**Pull request merged**